File: Python/0027. Calculator.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            s = input().strip()
            res = 0
            temp = 0
            power = 1
            for i in range(len(s)-1,-1,-1):
                c = s[i]
                if c != ' ':
                    if c == '+':
                        res += temp
                        temp = 0
                        power = 1
                    elif c == '-':
                        res -= temp
                        temp = 0
                        power = 1
                    else:
                        temp += power*int(c)
                        power *= 10
            res += temp
            print(res)
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0027. Calculator.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    string s;
    while(getline(cin,s)){
        int res = 0;
        int temp = 0;
        int power = 1;
        for(int i=s.size()-1;i>=0;--i){
            char c = s[i];
            if(c!=' '){
                if(c=='+'){
                    res += temp;
                    temp = 0;
                    power = 1;  
                }
                else if(c=='-'){
                    res -= temp;
                    temp = 0;
                    power = 1;
                }
                else{
                    temp += power*(c-'0');
                    power *= 10;
                }
            }
        }
        res += temp;
        cout<<res<<endl;
    }
    return 0;
}
